src/main/scala/org/apache/spark/sql/delta/commands/CreateDeltaTableCommand.scala

```diff
@@ -267,4 +282,60 @@ case class CreateDeltaTableCommand(
       }
     }
   }
+
+  private def getOperation(
+      tableVersion: Long,
+      metadata: Metadata,
+      isManagedTable: Boolean,
+      options: Option[DeltaOptions]): DeltaOperations.Operation = operation match {
+    case TableCreationModes.CreateTable if existingTableOpt.isDefined && query.isDefined =>
```

Comment:
Is this a valid case? I'm not sure this is really a CreateTable.

Reply:
This is old `saveAsTable` behavior